frontend/src/component/MenuList/NavItem/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { forwardRef, ForwardRefExoticComponent, RefAttributes, useEffect, useMemo } from 'react';
+import { Link } from 'react-router-dom';
+// assets
+import FiberManualRecordIcon from '@mui/icons-material/FiberManualRecord';
+import { ListItemButton, ListItemIcon, ListItemText, Typography } from '@mui/material';
+
+import { dispatch, useAppSelector } from '../../../store/hooks';
+import { viewsMiddleware, viewsSelector } from '../../../store/slices/menu';
+// types
```

Comment:
**nitpick:** I was not able to guess that some code from .../types can be not about types
[OMG](data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBxITEhUTEhIVFhUXFxUWFxUVFRUVFRIXFRUXFhUVFRUYHSggGBolHRUVITEhJSkrLi4uFx80OTQtOCgtLisBCgoKDg0OGxAQGy0lHR0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLf/AABEIAOAA4AMBIgACEQEDEQH/xAAcAAACAwEBAQEAAAAAAAAAAAAEBQIDBgEHAAj/xAA9EAABAwMCBAQDBQUHBQAAAAABAAIDBBEhBTEGEkFREyJhcTKBkQcUQqGxI1LB0fAVQ3KCsuHxCDNTYmP/xAAbAQACAwEBAQAAAAAAAAAAAAADBQIEBgEAB//EADERAAICAQMDAwIEBQUAAAAAAAECAAMRBBIhBTFBEyJRMmEUcYGhI0KRsfAGFSQz0f/aAAwDAQACEQMRAD8AxYsMD6qHi2uoxtQFZPa6+k1uqVbj2i/Ve99nxBdSr+iTvcSmMNGZDc7JgzS29ljuoax77DzxGul6e23OMTOhhVhpnDNlpG6Y3siv7JLm2AFlXop35Ugk+JZfRFRkmZGKAk7fNarhfQvN4z8Mbnm7kdAjdH4eaT5yA0ZPqfX0R+oakwYFhFGMNH94f5J5ptCtI3N9Q5/KZ6zUetd+HrP5nxM/xBUczy4n2Hos4+11fqdY57y47n8vQIDmSXWao2OftGqVpSoRR2jOjEXUkFcrZHbXQcV9/qj6NgcCD0/RUu8OpyIJFCXKbaO982PqmrIRa7SLjv19CqayoaQDy2PX0+fVdAkjXFsMZa4XGbpvPSxuAdflPUd/VKJJj3UDUOO5UgSJAbRwY1ETRs4Kb2MO4/r0STxSmFNIXDKu6dRcdvmDaxVE6Y2tNxsjoHg9EHLAVbSghO9FQ1VmGlLUWqy+2OoKfmNgOn6K7+zn9lZoEzee5/rCa1FY1sjNrE2Ptsnj2sGwo4xEFtti2bQJkq6nJxtus/NHYr0HWoGklw2ucLI6hSZuNkt6hp/Xr9QDmMNFqQyxT4aIpqUFckgIUI5yMJItdVbg2LGBJYe0wx9MOish04lSowXZKdxObZNaBpXOWHE9+HuK5E+kd5bpK+MveAm9W7BCD06LJKta720qv2kaF9bUnPzD4IABZEMaFGFiLjhWRCb7MCbHAReZKOAGwRkjA1oz6Lk8Voy7ayRmsJHK4rXaXRooDLwfMx3V9c92akPAPOIRUPe+7GYzk+iG1Ch5QG74ufZM9FDZCRazRa56u7j2sD9VLVC3xHd3NkA/ytAH8VT6xrRTV6Y7tBdK0r79xGAJ5vVu8x919EB1VlXCQ9w9Si6agJCyScxuKmZzxBWjNwEzhpTym2MJppGgk5IWxp9BbyjCnjiW6tMZ5m6J7e6HkY4r0+o4faQlk+ggdF4CEbTGefinN1P7stjNo/ohjpgRFQESI0cyzoFfSN5cp5Np4CXSQKzQDW4YQF2lGIQJh5gqX1A2Q8jCEK4OT7/cSRjESvptp5h9JWcrt0bNWZGbpK1hRLGFG0+qtA24gXqXOZqqKQPHmPdSrdKFrjLUggqLJ5S6r5Cx2xFkxDFuU/URbbVYjbq/6RLU0osbJBMyzlq5Iy32OxSrU6cFwPdU+p6JbEDJ4jHT24PMjQPwj+ZJ5HclrI6nnBCz1gNbbJoNJcrrCefmamFKxrWgJZQvvhXTxvuLFaLWWf8AHBAyTxFfTwfWLHsI6gZhFudytuldIHAWJRcrr7qppdCKDvsMu63X+shrq7yiWtLhYoCrFxdElijLygXKZJcgERjSFG9spp67wwAMYygavVC836jCBrKm5xshAViupurXHBziaTTEhBkQvk5nXTugg2SugZlabTILm6qJGVSzTaRTgAYTkNwgqBlgEW6RFh8SqZBTo57gUDVBSAkwIrqClkr0znSWrflESelE70DIwEq+RDyCysKcGU7iDKpYQq/BaV85yg16vaNwWwYn1YG3MvZTein4AsrItgpNzhalKkA7TPliYC+n7L6K4OU4bTAtJ6/1hEUumiVji3duf9kFq0U7gcSBvUD3Tmm8r/2b9jse10r1elLSWn8J+qYU4MbhzNtlGa4wSMEltrA26juvWDLFf5W/vALZsuHwf7zBVblTBOWlWagLOKDWK11hW4jyJoaSQMiPaKYh+FoqV0YYXH4+5/Ro6BZ/S4OZ/tkk7f8AKfPhGFrNIoavL/PEXWKzHYuefiERgubzXVU0h6qWOWwVAcSCDuqeqsdCSTlY1poCqNowRLA/CCriLFCyVT72B2VjWk2LiqDoXQmmERsN74mlgO6jAy5Wsj0prmEg57Inh3hEPPM847d1nLEKvgxlVXu5EV6XQk5stjpdERa4sAnlPpMbBZrQPVcDLFGA4jJVAE4wgCyi5TexSjspKJOCvNkLM9H1DQlkxRAJyA1UqRVT7lMq0pNOUfHEg5xK3yWS+pqrlFyvwltQxQYkciULTI+IvudCuNl2OVWNG/u5irVHKxxFP5fVFUMRccbjP0SpkouE3paixu05Wu09hcd+REVgI7SfjGxb3N0+4cYGgkncEEehWaLruJ9U2pKxEuQvXgSlqkZq8Dz3jSupwckdfqO6FoorufCdi02RLS7kBPXHshq6qa2WM9QM/wAlVTdjYP8AMSnRn6P84mC1eOzksKc64bvcelzb5lJysn1dcXt95rKDmsTSaI8XTpxystocnnWmid1V27XMaVA7CNem0rgkjmWz/ClP3stdbcL6uqSCbJaxpJuVTbVPY42y1dtUYEZvguea2CiqemJ6YS99SSQG7ALR05BaL9k0q6hVSdiDMoroTd7iZEVfIBG0AA/E/qB1Wi0/iGmbZodYAfVJnvpo2XmsXO+o7WWRrKmJz+WEOOcdz7BKeqMh1BCy3pj6aDcZ60zWWOF2m6+hjcQXled6Q6XBbcW3B3+i9DpgTHudlRDRkjhhxJv1KO1nEBKp9egjPx39lluIJnBxaCs++kfa5uu5YeIOyzaZ6G3X4X7OVUupxrIadUwtafEjeLbuGR8+yZR11I4WB+uCpK2e0iLgeIwqKhjhhJqz0UamJoJLXIZ0p2KOGyOZJnkQLqueHGFY1TspYgMAiIqiEhQgam9TEl4wUbSIA/MUa1NoyJSL3TakaQLqhkYtdFCXC0ejo9NiWMTWtu4EvjKIpxc29EtgqMoiJ9yUyrtV14gGQx3ValYNFtmi/uk8sxcS49VyvBu31aP4qL22aoooGceIOupaxx5ibVH3StG178oJYjqj77iY4pGFEN0qSzwtQ1+Fj6d1nArUh/kuh1nfVg+I20L4yJN8YO6g+IAJPLXG6kalyAX5wvaXPxCHuOYwiACY08uw74Wejq+6a6dJzPZb95v6hTpOXEkLl2HHxHfEmjl8jbHHLa3t1UuFuHKlszBKAIonFwJ5dz2Iyb36rZVulk2IPmQf3GU/E827DCt6tl9UmUqaBfWrZk6+jZzF7Wi+Mi21xe49kLqWrBjbDe1lKZwa3HT81ktVqSXKgBk5jhaxWmZXUtc93Mi9NgMvMxtg4DBOwXKN4c1F6XHyShw26ozLBMuTmWS8NXpZGMcTIXNkzbzFuOQfrlKdP4enfKZ6lth1Fmtviws0YHRei/cw9t2ktPQhKaulqtuYEerRdRrwO8q2aQOwYHEwtVpr2Ody/DfF1T4RvkLaHQ3HMjiT9AgpNOAk9FMd4YV44zEjKbC4WJ9VMa0YSGpejLJFcDMHlZdCOpQVZJNZdpnp505KnOCJnOp2MMYlMtmtVcL+YXOAFGsdc2XWMv5Qh67UstmxDwOIPRacONzCE0tXECA5hI62Nj8kdUQNY7yHma4czXWsSD3HQjI+SURUvmsU1cLNaD0v9FY6TqbbGKv/ACyGv0yoBYPM5K7AJ6bXVcEoeeS+Tsh5Wl5zsNkJTttJ81y/rAW9UX6c4MlX00mrce/iAakwteWncGyETPXh+0v3slazXUF2all+8lX9InQtHp8147JDMyxRmlTWwoBChKS1pbAHz8wmajBKubTABfSxncIdxk9UInaTiNCqg5xIyxNTfhtg8eIf/Rn+oJL4Tuqc8NNP3iIf+7fyyi6b/tX84EjIPHfiexiQILVJQ2NxG9sLsTsKnVYrxOP9bLtpyxMY01BECjxFFa6Pwbk+b8lj5m85wihDO6JwcDjY90j8CoYedoODtug7/tC2vwBjiHaXU+G/ldscey1VNDYrK6vFbldaxIB9iQtlw4C+Bjjvt9Cio2RiCAwcGO9PnLNxcJkHsOyHZBhUvmDV3zJ7AZLUCLLOVzgLo6uqSUgrXEqaiSFcBq6gpY4XRFQq41d0lQssCmUdbZtQ4gk1Mq42kDKOecqiqOB7rTfhK6ENg7gTJ2ahrnCn5iqoBvdW6e4g5Xah9yAEQ2LA7rH3HdYSDNFRXtH5Ri+IENP4g/l/ymO/5EfmhdQf5TnKvafyH5nCV17rustKzCjSG7ywAiizNmo9LwCTLKM4N1ClF3KINgoSzcjfUrL0rvsBPYcmOrbBXXANUk5noaJlzZce+5urqJt3BCssFuoLeIpUY4llUxUQP5SCjpWoJ8XVXNeoR90hTuxkTR0coIuimSN7LL09UWhEtriqTPHtOrUrzHM7mqzR5AKiL/GFnpKolWUNWWyscejmn6FcWz3ZnH1Ckz2yLorqh45CO4shYX3APdL+I9SEIFzYnK63eX0bMcUFHEGcryPmqNW06EMPIRey8/n4ljv+I+5VkPEDHYa5wPqb3UQy5hcV5+vmVVUPO6zzsVs9Ce1sQYOmyw1XPm4TjSdRvYKeODicbBPPebV0/lwUrqZb9VLxfKgZyuicBnXuQFUiJHWCBqZMIgns5imq6oNj8q6qcgWuyrWks23CL9dyphpchK84HurmFU1jCSOwWr1Ts2kYgc4mTqAGoGe2ZymgG6IEOVynYXkAW+aJmhLDa+Ui0XTHsb+IMD9471PUqahtQ5acFr26JHOS5xPqnkLQb37H69EFJGGtJKb9Q0frVCtThV5MTafVbLS7ckwCR/L7pdUOJVz5uYqicrNXIi1kL2EutaztlpQmeltG5S0BWNeUupRRyZPOIZOVW9twijDzk2USy26v1Ues+55O9ti4WBzR2CjTx3NldObrtGLOCjqagW9vYQVB5AMpkjIKiE/mow8KkaVlLcfEYtpzniej8I1ni08bjvblPu3CVfaFASI3dLEfxVfAxMZdH0+IfoVoOKqbnpz6ZRX5jGv6QDPGqrBVLXkbIzU4rFBBU+0T6jK2GO9Nr+byO36Jjp83LIAs1SwPc4cgJPotfHp7hyPcLHFwrVJJ7y9p7mcYbuJrIJPKFyUqMWw9lyVyJ2l8HiCTPQkyIkKGn2U5KJ6spcHZRtYUuaco1B/iCUNYeIxgTuGlDqZ9hkWPrbqkNO5arhhwIe05uCLLYlttAYeJhuoOUBf4MSadF5gr9Ww+3omsFCGuJQOvgc4PoirYGsGPiVUvWy8Y+IJSMuHHsFn9aqPwhaCmls13ss7Ww87lV6kzekVXu3EZaRSbScdooa9MKOnDzlVzUBGyK0ttjlZivT3VNtsHEbKytyIzp9JZ2R0WkR9lKmcjoTdWwB4hCJi4qoMUKmpDspaXLiVPrOfaJwE4wYT4oXzH+YFDBF07eYhCbUs67Z2pPcDNFTOuAiQgKU2wmDVDzHqDIhek1HJMw9L2PscLeVDOaIjfBXnVrZXodDLeAP7tB/JdEnjE8p4mpuXplLdA0szyBtvL1PZafiqMOce5JKd6LSNgp2Fu7hclR2gvBXaMvcD4xL9N0djTZoFm9UbXUoLcDZD6ZUXu4fi6Iusm5Y3k/u4+aIe8tCnEFjbgKqcIDSqwutfumVY3CkJDODF9roerwEXEwlA6sbBTnd0QVsiDYrZMlfNarWnTBzFOqfccS2IrQaBNyv8AcLOxbpnSyWIK0mif1UZfGJnuo6cemQfM1vPdJuIGfCUfRPuh9bZ5UaobbBM3p/ZcIic4AFBSAbhV6tVhreUblc09vMxB1JW+8VKeVGZqNO/pIWI+qRe9Ra5dlbYqsoTE5wZwHHIjKnqbJ3p0ocsoCi6WrLTuoGpXHEOt57NMuuri+WIlmfJnp0fVLEfR1dsFSWWNMVD8xuEbA5AQyg7Ij7wxoy4K3XTY/ZY39atBywhxW+oRy0kZO3LHf6heTz630aF7Dw/EZtPjIy50DCP8QAI/MKZTb3PMGmqR3ws851FvPO/l+HmNvmcLaaDpcklO1ha11wRcEc0ZabG4P6LLT6c7xS4AgXvbq072t6FOqHx4y57DyF17kva29+qr4IJ+Y3FTEEiOYNDliIBjDhmxJFsd1muLKw+GCSOdznXA2sLAW9MFXV9eQ281U57ujGPLh83bWWN1LUDI65+XYLvu7QNtnpqdxGftGGkzYHutXXDyA+ix+gxOccdwttrA5Ry9moh8SiGzzB9LZdpKVa2zdaDQYv2Dn9CbBJdaC6D7p4mZIsU2wojwMrpZZMK3HaU2QZyZUIQFcxRaphabp1WxcxF1B9xxNFpBuFTxa/kgJG9wiNCGAu6u3xJGQhvMT8Q6AHuoWgmwgHEytZA1WT2HJ/KeWyPLjnK0OmNsxV6jowiqXs6NOPYi6IdK1jVS6VpmpZrrD3mkutV1ATsZVVuBQhS2eqcScoinqz1Qj1Kq60jtJ+iyrL+ZSLlIw8wu0oCaR7TYheutNK7m7Hz4nlXd2ga61t9lyyKjlaBjdZqmgM2HOJdJ+Ja2gsLuNlxnhNOcqNe9wPKUEr1t9NLbaUHHkwSqxGWMMkrTs0WCHEmcqckQDQ4G9+nZUKndqLWPvP8A5CBAJJxF8bL3X7I64SUTG380ZcwjsL3b+RXhC9j+wzTnGGqqASQ10bQ32Bc930I+iEnOYel9riaviLhuOUE5a795uL+46rHS8CTk+WdtvUG69Sr23ZjslFJUDqoDniParCR3mAm+z1wF31A+TT/NBScLRMIAJcev/C9F1iYcqylOwveS48rBufREAx2k2UEZPeEcOaSxn7QizIxzOJ6u6NCUcQ13Nz2GXHA9zgK/XOJWPLKeKwiZlxH4rd13gyg+9VBneLQxG7QdnvG3yCljyZWJ5midS+DTRxdeUE+5yVktVOStVxFWi5ysHqdXupp3kSYFI9UPlyqecqIVuo4YGCbniFMKsCnpenzTu5II3PPWwwPc7Ba2i4LZEOerlF//ABRn/U7+S0dXUErAUcmJb9IWyScRXpUxtZgue/QLR6PQBp5nG7juUFLPGDaJga0bWTChnXL2ZlzjGZiepnazLX+p+Zi/tAYYpjJbDwM+oWFqKlztyvUvtMgDqUOtkOB+twV5KkHUtTbhawfbiaLoji3SqxHI4/pJNKa0JYRYjKUI/Te6h0q0i0Jjg/MZ3L7YVUeUXBshG1Yd8QuuV05JshAEfWa0i3bV2HceJGur25M4SuK57xygAZ6nuqUlbv3lgSRJKiiYZGhrgW3J2PZVthcdgfoiemxxtySfsZzIlS6jWaXKfw/VER6I/qQFZTpers7IYM3IPMVBe9/9OhvTVTSBy+K353ZYj8gvKKLRWFwaLvcTYNaNz7L9AfZbwtJQwv8AEAaZS13IMlthbzHuiarpzaWktaw3HsvnvPJYHPHiMqqjLHFhvyn4HfwWV1OTwnkPFv0K9MnhDxYjH9bLKa7SmP8A7rBJGcA28w9COvyS1TmNaNT4Peef6rrjAPiFlj9W4gJBa02BW31Lh3TpXEgFvoHED6JdHwjp7Tckut05ro4OJZNhPExOh6VLVSBjLhpPmd2HVesGSKmhbFHYWFsde590uFWyJvLBGGjv/ulLpnyOs1rpHHowE/UqfLdpDsOYNqtSXk2ykcwAPmyewyfot7pvAFVPmUiFnYZefdbXReCaWm8zWc7/AN5+T/spblTuYJrp5Fo3BdZVZZFyMP45fKPkNyttp32eUdOOapcZndvhZ9Bk/NegyRvOBgJfVUPddVwTgnA+0E1rEYEzdVqAY3w6eNsbR0a0N/RZ2eKSV1iTZaLVXtZgBI4/Ee6zRumunYV8qJWYb+8sg0+Ngzkrpjs4WFrrR6Nwk4+eY/JU6+I2ytYzoM/kurqw9m0HMRda0afhi44ImY4lpPFgezu029xkLxGRhBIO4wvfquoYxpc8gAb3Xkmp6cySR72m3M4kDpkrmo0FurQemOVi3/TuoKI6t28H7+Zn73Axt1Xxl6BWTscy7TshUmtd6iUxg9jNUMNzJ7rgKjdfIBfIHzO4jf8As5g3cVY2GIdCUG+dx3XOcrSizTof4dY/USptc92h3jsGzAu/fz0AHyS/mUuYqf4x/wCXj8hPemPMYw1TnHJWg4a0CorpPCp24FueQ/CwHqT1PokfC+kyVdRHTx/FIQL/ALo/E4+wyv1DpGnU2n0zY2lscbBlziBzHq5xO5KHqOr2U14Xlm/aeXTgnJ7RfwjwTTULQWjnltmVw8x9B+6FqQsJrH2saXBcCYyuHSJpdf8AzbLAcRfbjLI0so4PDv8A3jyC4ezRhZpjZa25zkn5lvgT1vijiiKjbkh0rvgjvk+p7BZ3guSeqqTUTkkBp5R+FpOAGjpi68p4RZNXVQMst3PN3PkdsB1z+i/Q2j0cUMTYo3AgdbjJ6k2UmQVr9zJgjEC1mCkkHJIxryTbyjzC/W4SODgClc42dO305hb5EtWvNMzm5rDG1ts7/NfU8Lwbl+M+W3fZQ3ADiSDkDiI6bgSiablr3/43kj6CwT2l0+KMWjja0ejQEWvlEsx7mRLE+ZEMC+5QpLhUZGVyEALP6tUnNk8dEShp9O5grFTKpyYQYHmYePTXzPwDY9VrtF4fjiFyLu7plS0bWCwVWp6rBTsL55WxtHVxA+g6qd2qaz2r2nC3xK9TqOVpb1IwvI9b1dsMj3yHuAOp9gpcZ/a3G+7KKMutf9q8WaPVrdyvK63UZJXF8jy5x/rA6Jr02sVqSw7xd1CgahAhPHn7xnq+tSVDvMbN6NG3z7quKXuk4e47K6Nrk80+oKn2iAFCooVRgCMKukbIM79CkdTpb27C49E6gcQiQUbVdOo1o3MMN8ieS56uPEyTG2Iv3zfC+qC3mPLstNU0bH7jPdKqnR3D4Tf06rPavod9Sn0/cP3lyvVI3fif/9k=)